File: binary_search.java
public class BinarySearch{
  public static void main(String args[]){
    
    int[] arr={5,12,16,24,32,39,43,56,76};
    if(binarySearch(arr,56)){
      System.out.println("Target element present in the array");
    }
    else{
      System.out.println("Target element not present in the array");
    }
  }
  public boolean binarySearch(int[] arr,int target){
    int s=0;
    int e=arr.length -1;
    while(s<=e){
      int mid=s+(e-s)/2;
      if(arr[mid]>target){
        e=mid-1;
      }
      else if(arr[mid]<target){
        s=mid+1;
      }
      else{
        return true;
      }
    }
    return false;
  }
}

File: BinarySearch.js
function binarySearch(arr, l, r, x){
  
    if (r >= l) {
        let mid = l + Math.floor((r - l) / 2);
        if (arr[mid] == x)
            return mid;
        if (arr[mid] > x)
            return binarySearch(arr, l, mid - 1, x);
        return binarySearch(arr, mid + 1, r, x);
    }
  
    return -1;
  
}
 


let arr = [ 2, 3, 4, 10, 40 ];
let x = 10;
let n = arr.length
let result = binarySearch(arr, 0, n - 1, x);

(result == -1) ? document.write( "Element is not present in array")
                   : document.write("Element is present at index " +result);
</script>
